src/index/IndexImpl.cpp

```diff
@@ -363,44 +388,54 @@ void IndexImpl::createFromFiles(
   // For the first permutation, perform a unique.
   auto firstSorterWithUnique =
       ad_utility::uniqueBlockView(firstSorter.getSortedOutput());
+  auto firstSorterWithUniqueIR{fromGenerator(std::move(firstSorterWithUnique))};
```

Comment:
more simple:
`auto firstSorterWithUnique = BlocksOfTriples(uniqueBlockView(getSortedOutput())`

then much less diff fruther below. Or even better, directly modify the `uniqueBlockView` s.t. it yields a type erased input range (e.g. by implementing the wrapper there).

Reply:
fromGenerator removed [here](https://github.com/ad-freiburg/qlever/pull/2151/commits/422819e4030ddc2d1f2111ee3145b76a9cefbebc)